ffs/manager/manager_test.go

```diff
@@ -23,30 +25,85 @@ func TestMain(m *testing.M) {
 
 func TestNewManager(t *testing.T) {
 	t.Parallel()
-	m, cls := newManager(t, tests.NewTxMapDatastore())
+	client, addr, _ := tests.CreateLocalDevnet(t, 1)
+	m, cls, err := newManager(client, tests.NewTxMapDatastore(), addr, false)
+	require.NoError(t, err)
+	require.NotNil(t, m)
+	defer require.NoError(t, cls())
+}
+
+func TestNewManagerMasterAddrFFSUseMasterAddr(t *testing.T) {
+	t.Parallel()
+	client, addr, _ := tests.CreateLocalDevnet(t, 1)
+	m, cls, err := newManager(client, tests.NewTxMapDatastore(), addr, true)
+	require.NoError(t, err)
+	require.NotNil(t, m)
+	defer require.NoError(t, cls())
+}
+
+func TestNewManagerNoMasterAddrNoFFSUseMasterAddr(t *testing.T) {
+	t.Parallel()
+	client, _, _ := tests.CreateLocalDevnet(t, 1)
+	m, cls, err := newManager(client, tests.NewTxMapDatastore(), address.Undef, false)
+	require.NoError(t, err)
 	require.NotNil(t, m)
-	defer cls()
+	defer require.NoError(t, cls())
+}
+
+func TestNewManagerNoMasterAddrFFSUseMasterAddr(t *testing.T) {
```

Comment:
Test to make sure creating the manager fails with no master address and use the master address for all ffs instances